main.py

```diff
@@ -124,9 +128,35 @@ def add_new_stream(self, *args, stream_url=None, stream_quality=cfg[CONFIG_QUALI
 
         # Run the rest on a separate thread to be able to show the loading feedback
         # Also helps a lot with lag
-        threading.Thread(target=self._add_new_stream, args=(stream_url, stream_quality)).start()
+        threading.Thread(target=self._add_new_stream, args=(stream_url, stream_quality, 0)).start()
+
+    def add_new_scheduled_stream(self, *args, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
+        """Schedules a new stream at given time"""
+        if not stream_url:
+            stream_url, ok1 = QtWidgets.QInputDialog.getText(
+                self,
+                "Scheduled stream",
+                "Enter the stream URL:"
+            )
+            inputTime, ok2 = QtWidgets.QInputDialog.getText(
+                self,
+                "Scheduled stream",
+                "Time (HH.MM)"
+            )
 
-    def _add_new_stream(self, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
+        try:
+            h, m = inputTime.split(".")
+            now = datetime.now()
+            runtime = now.replace(hour=int(h), minute=int(m), second=0, microsecond=0)
+            if not runtime > now:
+                raise ValueError
+            delay = (runtime - now).total_seconds()
+            QtWidgets.QMessageBox.information(self, "Scheduled stream", "Succesfully scheduled stream at " + h + "." + m)
+            threading.Thread(target=self._add_new_stream, args=(stream_url, stream_quality, delay)).start()
```

Comment:
Why not just create a QTimer that invokes `_add_new_stream` after `delay` seconds? Then we do not need to add a parameter to `_add_new_stream`.